flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/CatalogLoader.java

```diff
@@ -44,17 +46,27 @@
    *
    * @return a newly created {@link Catalog}
    */
-  Catalog loadCatalog();
+  String getName();
+  Map<String, String> getOptions();
+  Object getConf();
```

Comment:
The interface for `setConf` that avoids needing hadoop is found here: https://github.com/apache/iceberg/blob/da84a9c0740c9909889780b4350e113cfd35a4ae/core/src/main/java/org/apache/iceberg/hadoop/Configurable.java#L22-L27

We might just want to add `getConf` to that as well.

Reply:
IMO there are two independent questions we want to answer at runtime.

1) Can the runtime environment provide Hadoop configuration?

2) Does the catalog accept Hadoop configuration?

The existing Configurable interface addresses the second question.  But we still need Flink to answer the first one.  In an ideal case, Flink would directly provide an answer via the Context param somehow.

I also don't think adding getConf to the Iceberg Configurable interface really gets us anywhere.

Reply:
I also don't think adding getConf to the Iceberg Configurable interface really gets us anywhere. 

> IMO there are two independent questions we want to answer at runtime.
> 
> 1. Can the runtime environment provide Hadoop configuration?
> 2. Does the catalog accept Hadoop configuration?
> 
> The existing Configurable interface addresses the second question. But we still need Flink to answer the first one. In an ideal case, Flink would directly provide an answer via the Context param somehow.
> 
> I also don't think adding getConf to the Iceberg Configurable interface really gets us anywhere.

you are right